### PR TITLE
Topology v2

### PR DESCRIPTION
**Pull request by latot**
Improve explanation of basic types in topology.

And include geometry representation.

### CODE REVIEW

File: postgis-intro/sources/en/topology_base_types.rst [hunk 1]
```diff
@@ -0,0 +1,186 @@
...
+Is good to know, until now Postgis will save all this information internally, each type has its own unique ID, and with it we can edit and change them using the available functions, for example with ST_RemEdgeNewFace you can remove an edge and create a new face.
+
+But there is still a missing piece, how do we say for example, a face has a specific attribute? Postgis implements several concepts to do this like Layers, TopoGeometry, TopoElement.
+
+Universal Face
+--------------
+
+Could be intuitive to think that only exists the faces constructed by edges, there is one exception, all the white space out of all the faces are also a face!
+
+The empty space is called the Universal Face, this face constains everything, each node, edge and face are inside the Universal Face.
```
Comment (strk) [inline postgis-intro/sources/en/topology_base_types.rst:40]: This is not how I see it. The non-universal faces are (in my views) like "holes" in the Universal face, so they are not "contained" in it as they take up space which is NOT part of the Universal face. The Universal Face is the infinite set of point that are not on the boundary or in the interior of any other face.

File: postgis-intro/sources/en/topology_base_types.rst [hunk 2]
```diff
@@ -0,0 +1,186 @@
...
+  :align: center
+   
+The Universal Face has the ID 0.
+
+Edge interpretation
+-------------------
+
+To correctly represent a topology and their forms, there are some definitions that are used to construct the tables that store everything, the edge is on the more complex ones.
+
+All the information about edges are stored on the edge_data table, in your custom topology schema, and which information do we need in the edges? basically its nodes and faces information, edge is the primitive who connects both.
```
Comment (strk) [inline postgis-intro/sources/en/topology_base_types.rst:54]: I'm not sure I'd document `edge_data` in the workshop, as it is mostly an implementation detail. The important thing to document is the `edge` view, which is also defined by the standard.

File: postgis-intro/sources/en/topology_base_types.rst [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+
+And their interpretation:
+
+- abs_next_right_edge: Looking the edge forward, which is the next edge who build the face on the right side?
+- abs_next_left_edge: Looking the edge forward, which is the next edge who build the face on the left side?
+
+Next Edge
+?????????
+
+This columns, next_right_edge and next_left_edge are very similar to the abs ones, actually it follows the same rules, we get the next edge id for the left and right faces, and when we have them we perform one extra operation.
```
Comment (strk) [inline postgis-intro/sources/en/topology_base_types.rst:124]: The standard only defines these 2 fields (next right, next left). The sign of that value tells us if the next edge is encountered in its forward direction (positive id) or backward direction (negative id). Encountering the next edge can be described as walking on the side of the current edge and ending up on the side of the next edge. We always walk on the left side of the edge following its direction and backward on its right side, so you could also see the sign as representing which side of the next edge we encounter (left side if positive, right side if negative). An isolated single edge will have negative self as next left and postiive left as next right.

**Review by strk — COMMENTED** [review]
It's great to see this work, keep it up !

File: postgis-intro/sources/en/topology_base_types.rst (hunk 1 above)
Comment (latot) [reply postgis-intro/sources/en/topology_base_types.rst:40]: You are right, I try to explain it more easy, but is technically wrong, let me check how to write it again.

File: postgis-intro/sources/en/topology_base_types.rst (hunk 2 above)
Comment (latot) [reply postgis-intro/sources/en/topology_base_types.rst:54]: mm, this docs are mainly focused on the implementations, how to interpret, to be able how to use.

Topology is harder than SF, and does not has a lot of operation, any operation we want to create even for personal use we can need this information.

even on the start I wrote:

```
In this workshop, we will check some fundamentals of topology, its basics and definitions. It is not intended to be a way to use them directly, is to understand it to then be able to use it easily.
```

I'll write clearly the edge view.

Comment (latot) [reply postgis-intro/sources/en/topology_base_types.rst:54]: In the Topology Hierarchy I changed how I wrote it, has first a section closer to the User, and then more specific aspects of the concepts.

File: postgis-intro/sources/en/topology_base_types.rst (hunk 3 above)
Comment (latot) [reply postgis-intro/sources/en/topology_base_types.rst:124]: Thx, I checked this again, finally I understand what this does, glad you notice it.

**Comment by latot** [comment]
@strk The documentation is ready!

**Pull request merged**